### DISCUSSION TITLE
Is there a way to retrieve unknown properties from a data object after creation?

### DISCUSSION THREAD

**Discussion by jameshulse** [spatie/laravel-data#605] [Q&A]
We are receiving data from an external API. I have modelled data classes to match some of the fields that we care about, but in some cases there may be extra fields on the response that we haven't added to the constructor.

Is there an existing way to have these other fields be saved to the resulting data instance?

I'll try and give a trivial example. Here is a data object:

```php
class MyResponse extends Data
{
    public function __construct(
        public string $knownField,
    ) {
    }
}
```

And the response that we are receiving from an external API:

```json
{
    "knownField": "foo",
    "unknownField": "bar"
}
```

Is it possible to see `unknownField` when I call `->toArray()` on my data object?

**Comment by rubenvanassche** [comment]
Mhh I don't think so, why not define all the properties of the API?

**Reply by jameshulse** [reply]
Because I need to always pass the exact API response I received back to the API as a request, but with some fields changed. If the API added a field without our knowledge then it would be lost when converting to a data object. We selected not to use the data package for this particular use-case. Thanks though!
